libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java

```diff
@@ -491,6 +536,18 @@ public void onAuthenticationFailed() {
         }
     }
 
+    @TargetApi(VERSION_CODES.Q)
```

Comment:
This is the main point of the PR.  Biometric Manager is new in API 29.

Reply:
Add a `TODO:` here to remove this conditional if `minApi >= 29`.